Source/SwiftLintBuiltInRules/Rules/Idiomatic/FileNameRule.swift

```diff
@@ -57,6 +59,16 @@ struct FileNameRule: OptInRule, SourceKitFreeRule {
     }
 }
 
+// MARK: - For `excluded` option
+private extension FileNameRule {
+    func shouldExclude(filePath: String) -> Bool {
+        configuration.excluded.compactMap {
+            let regex = try? RegularExpression(pattern: "^\($0)$")
```

Comment:
We could make the creation of regular expressions part of the configuration (`@ConfigurationElement` have a `postprocessor` argument) that can be used for that. By creating the regex in the configuration parsing process already, we ensure that they are correct and can report errors. Here, we would just ignore the exclude if regex creation fails.

Reply:
That's a very valid example. You are right. The method should compare strings first to be backwards compatible and only afterwards try to match them as regular expressions.

And if the conversion to regex fails, we just assume it isn't supposed to
be a regex and just ignore it. There isn't more we can do.

Note that you can now also use the `Regex` library from Swift since macOS 13 is the lowest platform we support.